daemons/execd/remoted_pidone.c

```diff
@@ -156,11 +157,17 @@ load_env_vars(void)
          */
         value_end = end;
 
+        // Strip trailing comment beginning with '#'
+        comment = strchr(end, '#');
+        if (comment != NULL) {
+            *comment = '\0';
+        }
+
         while (isspace(*end)) {
             end++;
         }
 
-        if ((*end != '\0') && (*end != '#')) {
+        if (*end != '\0') {
```

Comment:
This kind of function is the perfect candidate for unit tests, though it's hiding somewhere inaccessible at the moment and I would hate to expose it even as private API.

Reply:
Yeah, we could make it library-private for that purpose. It is really annoying to me that static functions can't be unit-tested by cmocka AFAICT.

Reply:
Yeah that's just an artifact of symbols not being visible if they're static.  If you really want to test a static function, you can wrap the `static` bit in `#if !defined(PCMK__UNIT_TESTING)`.  We could even go so far as to define something like `PCMK__STATIC` (man I wish we had namespacing) to do that for us.  libcrmcommon at least is already set up to build a whole separate version of the library for unit testing, so we can do whatever we want.  This would be harder with daemons.